rfc/2015-06-10-message-api.markdown

```diff
@@ -0,0 +1,69 @@
+---
+title: API for messages
+category: rfc
+status: proposal
+---
+
+This RFC describe the CaliOpen message API requirements to implement current
+mockups.
+
+## Resources
+
+> NOTE the field names and structure are here to describe pieces of information.
+> Feel free to provide better ones.
+
+> NOTE the term **Link** is used for nested resources. The implementation can
+> embed the information OR provide information to retrieve it later (id,
+> href...).
+>
+> If the information is not embedded, then there should be an API endpoint to
+> retrieve missing information in batch and prevent too many requests.
+
+Message related resources are **Thread** and **Message**.
+
+A **Thread** is a group of **Messages** with additional pieces of information.
+Every **Thread** MUST provide the following information:
+
+* `total_count`: the number of messages
+* `unread_count`: the number of unread messages
+* `subject`: the thread subject. Empty string if none.
+* `contacts`: a list of links to contacts involved in the conversation
+* `importance_level`: the importance level associated to message
+* `privacy_index`: the privacy index associated to message
+* `messages`: a list of links to most recent messages. **<--- This is used for
```

Comment:
at this time only complex data left to compute is unread_count. thread.text will contain last shortened message (ake the headline, was previously slug in my response).

Thread.messages exist as a list of message's uuid inside this thread (so total_count can be computed), also threads.attachments for a list of all attachments exist.

Have one or two more commits to fit that (so unread_count is the only information in the current RFC I can't get correctly and something to discuss around message.status that I modelize today as message.flags structure)